File: 백준/Silver/2210. 숫자판 점프/숫자판 점프.java
import java.io.*;
import java.util.*;

public class Main {
    static int[][]board = new int[5][5];
    static HashSet<String> uoniqueNumber = new HashSet<>();
    static int[] dx = {-1,1,0,0};
    static int[] dy = {0,0,-1,1};
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        for(int i = 0;i<5;i++){
            String[] inputs = br.readLine().split(" ");
            for(int j = 0;j<5;j++){
                board[i][j] = Integer.parseInt(inputs[j]);
            }
        }
        for(int i = 0;i<5;i++){
            for(int j = 0;j<5;j++){
                dfs(i,j,"",0);
            }
        }
        System.out.println(uoniqueNumber.size());
    }
    static void dfs(int x, int y, String num, int depth){
        if(depth==6){
            uoniqueNumber.add(num);
            return;
        }
        for(int i = 0;i<4;i++){
            int nx = x+dx[i];
            int ny = y+dy[i];

            if(nx>=0 && nx<5&& ny>=0 && ny<5){
                dfs(nx,ny,num+board[nx][ny],depth+1);
            }
        }
    }
    
}
    

File: 백준/Silver/2210. 숫자판 점프/숫자판 점프.py
import sys
input = sys.stdin.readline

board = [list(map(int,input().split()))for _ in range(5)]
unique_number = set()

dx = [-1,1,0,0]
dy = [0,0,-1,1]

def dfs(x,y,num):
    if len(num)==6:
        unique_number.add(num)
        return
    for i in range(4):
        nx,ny = x+dx[i],y+dy[i]
        if 0<= nx<5 and 0<=ny<5:
            dfs(nx,ny,num+str(board[nx][ny]))
for i in range(5):
    for j in range(5):
        dfs(i,j,str(board[i][j]))
print(len(unique_number))

